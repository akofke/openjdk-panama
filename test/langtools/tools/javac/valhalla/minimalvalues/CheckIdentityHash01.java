/*
 * @test /nodynamiccopyright/
 * @summary Value types do not support identityHashCode
 * @modules jdk.incubator.mvt
 * @compile/fail/ref=CheckIdentityHash01.out --should-stop:ifError=PARSE -XDrawDiagnostics -Werror -Xlint:values  CheckIdentityHash01.java
 */

import static java.lang.System.*;
@jdk.incubator.mvt.ValueCapableClass
final class CheckIdentityHash01 {
    void test(CheckIdentityHash01 v) {

        identityHashCode(v);      // <- error
        identityHashCode(this);   // <- error

        System system = null;
        system.identityHashCode(v);      // <- error
        system.identityHashCode(this);   // <- error

        System.identityHashCode(v);      // <- error
        System.identityHashCode(this);   // <- error

        java.lang.System.identityHashCode(v);    // <- error
        java.lang.System.identityHashCode(this); // <- error
    }
}
