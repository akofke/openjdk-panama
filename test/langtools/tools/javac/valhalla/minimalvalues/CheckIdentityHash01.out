- compiler.warn.incubating.modules: jdk.incubator.mvt
- compiler.err.warnings.and.werror
CheckIdentityHash01.java:13:25: compiler.warn.value.does.not.support: identityHashCode
CheckIdentityHash01.java:14:25: compiler.warn.value.does.not.support: identityHashCode
CheckIdentityHash01.java:17:32: compiler.warn.value.does.not.support: identityHashCode
CheckIdentityHash01.java:18:32: compiler.warn.value.does.not.support: identityHashCode
CheckIdentityHash01.java:20:32: compiler.warn.value.does.not.support: identityHashCode
CheckIdentityHash01.java:21:32: compiler.warn.value.does.not.support: identityHashCode
CheckIdentityHash01.java:23:42: compiler.warn.value.does.not.support: identityHashCode
CheckIdentityHash01.java:24:42: compiler.warn.value.does.not.support: identityHashCode
1 error
9 warnings
